### PR TITLE
Update status definition to be more intuitive and promote some components

### PR DESCRIPTION
**Pull request by srt32**
Update statues to match the change made in primer/doctocat#217

Closes https://github.com/primer/view_components/issues/203

### CODE REVIEW

File: app/components/primer/component.rb [hunk 1]
```diff
@@ -10,15 +10,14 @@ class Component < ViewComponent::Base
 
     # sourced from https://primer.style/doctocat/usage/front-matter#status
     STATUSES = {
+      alpha: :alpha,
+      beta: :beta,
+      stable: :stable,
       deprecated: :deprecated,
```
Comment (srt32) [inline app/components/primer/component.rb:16]:
```suggestion
      deprecated: :deprecated
```

File: app/components/primer/spinner_component.rb [hunk 2]
```diff
@@ -34,5 +34,10 @@ def initialize(size: DEFAULT_SIZE, **system_arguments)
       @system_arguments[:viewBox] = "0 0 16 16"
       @system_arguments[:fill] = :none
     end
+
+    def self.status
+      Primer::Component::STATUSES[:beta]
+    end
+
```
Comment (srt32) [inline app/components/primer/spinner_component.rb:41]:
```suggestion
```

**Review by joelhawksley — APPROVED** [review]
Looks good! Perhaps we wait to land this until we reach a resolution on the dotocat PR?

**Comment by srt32** [comment]
ugh, our first auto merge fail.

**Comment by joelhawksley** [comment]
> ugh, our first auto merge fail.

It's all good!

**Pull request merged**